src/components/legend/draw.js

```diff
@@ -576,11 +577,12 @@ function computeTextDimensions(g, gd) {
     }
     else {
         var text = g.selectAll('.legendtext'),
-            textSpans = g.selectAll('.legendtext>tspan'),
-            textLines = textSpans[0].length || 1;
+            textSpans = text.selectAll('tspan.line'),
+            textLines = textSpans.size() || 1,
+            textNode = text.node();
```

Comment:
sure, that's a bit cleaner -> https://github.com/plotly/plotly.js/pull/1792/commits/27b4be3998b4580446efccde3f3e93183e4e7c6b